pkg/generate/app/app.go

```diff
@@ -81,6 +85,28 @@ func nameFromGitURL(url *url.URL) (string, bool) {
 	}
 	// from path
 	if name, ok := git.NameFromRepositoryURL(url); ok {
+
+		if !kvalidation.IsDNS1123Label(name) {
+			name = strings.ToLower(name)
+
+			// remove everything except [-0-9a-z]
+			name = invalidNameCharactersRegexp.ReplaceAllString(name, "")
```

Comment:
>There should be a function somewhere else that already does this ...

@smarterclayton Yes, I've found it and moved checks to that method (some of them already was there).

But. In this method, when we are checking for a max length, we are using `kvalidation.DNS1123SubdomainMaxLength` (= 253) but I believe that we should use `kvalidation.DNS1123LabelMaxLength` (= 63), because the latter will be used for name validation (see original error message). Would you mind if I update this check?